qp_shotgun/qc_filter/tests/test_qc_filter.py

```diff
@@ -0,0 +1,311 @@
+                       'Number of threads to be used': '1'
+        }
+        self._clean_up_files = []
+
+    def tearDown(self):
+        for fp in self._clean_up_files:
+            if exists(fp):
+                if isdir(fp):
+                    rmtree(fp)
+                else:
+                    remove(fp)
+
+    def test_make_read_pairs_per_sample_match_fwd_rev(self):
+        fd, fp = mkstemp()
+        close(fd)
+        with open(fp, 'w') as f:
+            f.write(MAPPING_FILE)
+        self._clean_up_files.append(fp)
+
+        fwd_fp = ['./folder/s3_S013_L001_R1.fastq.gz',
+                  './folder/s2_S011_L001_R1.fastq.gz',
+                  './folder/s1_S009_L001_R1.fastq.gz']
+
+        rev_fp = ['./folder/s3_S013_L001_R2.fastq.gz',
+                  './folder/s2_S011_L001_R2.fastq.gz',
+                  './folder/s1_S009_L001_R2.fastq.gz']
+
+        exp = [('s1', 'SKB8.640193', './folder/s1_S009_L001_R1.fastq.gz',
+                './folder/s1_S009_L001_R2.fastq.gz'),
+               ('s2', 'SKD8.640184', './folder/s2_S011_L001_R1.fastq.gz',
+                './folder/s2_S011_L001_R2.fastq.gz'),
+               ('s3', 'SKB7.640196', './folder/s3_S013_L001_R1.fastq.gz',
+                './folder/s3_S013_L001_R2.fastq.gz')]
+
+        obs = make_read_pairs_per_sample(fwd_fp, rev_fp, fp)
+
+        self.assertEqual(obs, exp)
+
+    def test_make_read_pairs_per_sample_match_fwd_only(self):
+        # do we want to support forward-only yet?
+        fd, fp = mkstemp()
+        close(fd)
+        with open(fp, 'w') as f:
+            f.write(MAPPING_FILE)
+        self._clean_up_files.append(fp)
+
+        fwd_fp = ['./folder/s3_S013_L001_R1.fastq.gz',
+                  './folder/s2_S011_L001_R1.fastq.gz',
+                  './folder/s1_S009_L001_R1.fastq.gz']
+
+        rev_fp = []
+
+        exp = [('s1', 'SKB8.640193', './folder/s1_S009_L001_R1.fastq.gz',
+                None),
+               ('s2', 'SKD8.640184', './folder/s2_S011_L001_R1.fastq.gz',
+                None),
+               ('s3', 'SKB7.640196', './folder/s3_S013_L001_R1.fastq.gz',
+                None)]
+
+        obs = make_read_pairs_per_sample(fwd_fp, rev_fp, fp)
+
+        self.assertEqual(obs, exp)
+
+    def test_make_read_pairs_per_sample_match_fwd_rev_diffnum(self):
+        fd, fp = mkstemp()
+        close(fd)
+        with open(fp, 'w') as f:
+            f.write(MAPPING_FILE)
+        self._clean_up_files.append(fp)
+
+        fwd_fp = ['./folder/s3_S013_L001_R1.fastq.gz',
+                  './folder/s2_S011_L001_R1.fastq.gz',
+                  './folder/s1_S009_L001_R1.fastq.gz']
+
+        rev_fp = ['./folder/s4_S013_L001_R2.fastq.gz',
+                  './folder/s2_S011_L001_R2.fastq.gz',
+                  './folder/s1_S009_L001_R2.fastq.gz']
+
+        with self.assertRaises(ValueError):
+            make_read_pairs_per_sample(fwd_fp, rev_fp, fp)
+
+    def test_make_read_pairs_per_sample_match_fwd_rev_notmatch(self):

```

Comment:
If these functions are exactly identical to those in qc_trim, just import them directly rather than duplicating code.